tests/Microsoft.Bot.Builder.Ai.QnA.Tests/QnAMakerTests.cs

```diff
@@ -2,31 +2,61 @@
 // Licensed under the MIT License.
 
 using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Bot.Builder.Tests;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using RichardSzalay.MockHttp;
 
 namespace Microsoft.Bot.Builder.Ai.QnA.Tests
 {
     [TestClass]
     public class QnAMakerTests
     {
-        public readonly string knowlegeBaseId = TestUtilities.GetKey("QNAKNOWLEDGEBASEID");
-        public readonly string endpointKey = TestUtilities.GetKey("QNAENDPOINTKEY");
-        public readonly string hostname = TestUtilities.GetKey("QNAHOSTNAME");
+        public readonly string knowlegeBaseId = TestUtilities.GetKey("QNAKNOWLEDGEBASEID") ?? "dummy-id";
+        public readonly string endpointKey = TestUtilities.GetKey("QNAENDPOINTKEY") ?? "dummy-key";
+        public readonly string hostname = TestUtilities.GetKey("QNAHOSTNAME") ?? "https://dummy-hostname.azurewebsites.net/qnamaker";
+
+
+        private string GetRequestUrl()
```

Comment:
I would suggest moving all these helper methods to the bottom (under all the actual test methods).